.github/workflows/start-minor-release.yml

```
@@ -0,0 +1,57 @@
+---
+name: "Start minor release"
+
+on: # yamllint disable-line rule:truthy
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "The Determined minor version to release. E.g. 0.38.0. This will create a new release branch and make commits on main."
+        required: true
+
+jobs:
+  start-minor-release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: "Validate version"
+        shell: bash {0}
+        run: |
+          grep -E -o '[0-9]+\.[0-9]+\.0' <(printf "%s" '${{ github.event.inputs.version }}')
+
+          ret=$?
+          if [[ $ret != 0 ]]; then
+            echo '::error::Version string must match <[0-9]+\.[0-9]+\.0>. Got: <${{ github.event.inputs.version }}>'
+            exit $ret
+          fi
```

Since you're setting the shell to bash, you can technically just use bash pattern matching.  Just think of the amazing speed improvement when you don't need a subshell and a fork!
```suggestion
          shopt -s extglob
          v='${{ github.event.inputs.version }}'
          if [[ "$v"  != @(+(0-0).+(0-9).0) ]]; then
            echo '::error::Version string must match <[0-9]+\.[0-9]+\.0>. Got: <'"$v"'>'
            exit 1
          fi
```

Feel free to ignore this suggestion completely.  People actually understand grep. :D

---

Huh. Neat! I always learn something new about Bash from you.